gwpy/io/utils.py

```diff
@@ -98,6 +100,50 @@ def gopen(name, *args, **kwargs):
     return fobj
 
 
+def with_open(func=None, mode="r", pos=0):
+    """Decorate a function to ensure the chosen argument is an open file
+
+    Parameters
+    ----------
+    func : `callable`
+        the function to decorate
+
+    mode : `str`, optional
+        the mode with which to open files
+
+    pos : `int`, optional
+        which argument to look at
+
+    Examples
+    --------
+    To ensure that the first argument is an open read-only file, just use
+    the decorator without functional parentheses or arguments:
+
+    >>> @with_open
+    >>> def my_func(pathorfile, *args, **kwargs)
+    >>>     ...
+
+    To ensure that the second argument (position 1) is a file open for writing:
+
+    >>> @with_open(mode="r", pos=1)
+    >>> def my_func(stuff, pathorfile, *args, **kwargs)
+    >>>     stuff.write_to(pathorfile, *args, **kwargs)
```

Comment:
This example is a bit confusing: which is readable (`pathorfile`?) and which is writable(`stuff`?)? Perhaps give a bit more explanation here for non-experts.

Reply:
It looks like mode in the example should be `w`, or `rw`, not `r` (unless we don't understand something fundamental...)